websites/O/OK/presence.ts

```diff
@@ -0,0 +1,203 @@
+const presence = new Presence({
+	clientId: "1036732879725658213",
+});
+
+enum Assets {
+	friends = "Друзья",
+	friendRequests = "Друзья",
+	pymk = "Друзья",
+	outgoingFriendRequests = "Друзья",
+	photos = "Фото",
+	statuses = "Заметки",
+	groups = "Группы",
+	market = "Товары",
+	music = "Музыка",
+	games = "Игры",
+	video = "Видео",
+	giftsFriend = "Подарки",
+	about = "О себе",
+	feed = "новости",
+	guests = "гостей",
+	marks = "события",
+	messages = "сообщения",
+	vitrine = "игры",
+	gifts = "подарки",
+	discovery = "рекомендации",
+	topphoto = "фото-конкурс",
+	marathons = "марафоны",
+	services = "приложения",
+	mall = "товары",
+	payments = "платежи и подписки",
+	online = "людей на сайте",
+}
+
+async function getStrings() {
+	return presence.getStrings({
+		playVideo: "general.watchingVid",
+		playMusic: "general.playing",
+		pause: "general.paused",
+	});
+}
+
+let video = {
+		duration: 0,
+		currentTime: 0,
+		paused: true,
+	},
+	strings: Awaited<ReturnType<typeof getStrings>>;
+
+function textContent(tags: string) {
+	return document.querySelector(tags)?.textContent;
+}
+
+function typeContent(string: string) {
+	return Assets[string as keyof typeof Assets];
+}
+
+function getMillisecondsFromString(timeString: string): number {
+	const parsedText = timeString?.split(":");
+	return (Number(parsedText[0]) * 60 + Number(parsedText[1])) * 1000;
+}
+
+presence.on(
+	"iFrameData",
+	(data: { duration: number; currentTime: number; paused: boolean }) => {
+		video = data;
+	}
+);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			details: "Где-то на сайте",
+			largeImageKey: "https://i.imgur.com/Jky2SvM.png",
+		},
+		[musicMode, privacy, time, logo] = await Promise.all([
+			presence.getSetting<boolean>("musicMode"),
+			presence.getSetting<boolean>("privacy"),
+			presence.getSetting<boolean>("time"),
+			presence.getSetting<boolean>("logo"),
+		]),
+		{ pathname } = document.location,
+		playMusic = document
+			.querySelector("music-mini-player")
+			?.hasAttribute("playing");
+
+	function showMusic() {
+		presenceData.details = textContent(".mini-player_name");
+		presenceData.state = textContent(".mini-player_artist");
+		presenceData.smallImageKey = playMusic ? "play" : "pause";
```

Comment:
Imgur links are preferred over Discord asset names. Also, move to an enum.

Reply:
> Imgur links are preferred over Discord asset names. Also, move to an enum.

I need to load smallImageKey "play" and "pause" to imgur, right?

Reply:
That's correct.